frontend/src/simulation/simulation.js

```diff
@@ -0,0 +1,247 @@
+                total += this.props.population[j]["traits"][this.state.categories[i]]
+                //Above: index into the population to get a person, then that person's traits and
+                //then the value (true or false) of that trait
+            }
+            aggregate_values[[this.state.categories[i]]] = total/this.props.population.length;
+        }
+
+        return (
+            <table className="aggregate_data_table">
+                <thead>
+                    <tr>
+                        <th className="aggregate_data_table_header">
+                            Trait
+                        </th>
+                        <th className="aggregate_data_table_data">
+                            Percentage of Population
+                        </th>
+                    </tr>
+                </thead>
+                <tbody>
+                    {this.state.categories.map((category, key) => {
+                        return (
+                            <tr key={key} className="aggregate_data_table_rows">
+                                <td className="aggregate_data_table_data">
+                                    {category}
+                                </td>
+
+                                <td className="aggregate_data_table_data">
+                                    {(aggregate_values[category]*100).toFixed(1)}%
+                                </td>
+                            </tr>
+                        )
+                    })}
+                </tbody>
+            </table>
+        )
+    }
+}
+AggregateData.propTypes = {
+    population: PropTypes.array,
+};
+
+class MainView extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            population: null,
+        }
+        this.csrftoken = getCookie('csrftoken');
+    }
+
+    async componentDidMount() {
+        try {
+            const population = await fetch('/api/population/');
+            const json = await population.json();
+            this.setState({population: json["citizen_list"]});
+        } catch (e) {
+            console.log(e);
+        }
+    }
+
+    render() {
+        if (this.state.population) {
+            return (
+                <>
+                    <div className="row">
+                        <div className="col-md-8 col-lg-4 col-sm-12 data_container">
+                            <AggregateData
+                                population={this.state.population}
+                            />
+                        </div>
+                    </div>
+
+                    <div className="row">
+                        <div className="col-md-8 col-lg-5 col-sm-12 budget_container">
+                            <Budget
+                                population={this.state.population}
+                            />
+                        </div>
+                    </div>
+                </>
+            )
+        } else {
+            return (
```

Comment:
I prefer to stick this kind of loading thing at the top of the render func --

```
if (!this.state.population) {
    return (
        <div>Loading!</div>
    )
}
```

-- bc then you can get on with the rest of the render function without having it all wrapped in a conditional